apis/Google.Cloud.Spanner.Data/Google.Cloud.Spanner.Data.CommonTesting/SpannerTestDatabaseBase.cs

```diff
@@ -0,0 +1,224 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License").
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at 
+//
+// https://www.apache.org/licenses/LICENSE-2.0 
+//
+// Unless required by applicable law or agreed to in writing, software 
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and 
+// limitations under the License.
+
+using Google.Api.Gax;
+using Google.Api.Gax.ResourceNames;
+using Google.Cloud.Spanner.Admin.Database.V1;
+using Google.Cloud.Spanner.Admin.Instance.V1;
+using Google.Cloud.Spanner.Common.V1;
+using Google.Cloud.Spanner.V1.Internal.Logging;
+using Grpc.Core;
+using System;
+
+namespace Google.Cloud.Spanner.Data.CommonTesting;
+
+/// <summary>
+/// Base class for Spanner Test database.
+/// </summary>
+public abstract class SpannerTestDatabaseBase
+{
+    protected static readonly object s_lock = new object();
```

Comment:
The lock shouldn't be shared across derived types.

Reply:
This is a static declaration, which means there's exactly one value for it, it's not inherited, and it doesn't exist per instance.

If you have class A and class B that extends SpannerTestDatabaseBase, whe you use the lock in class A, it cannot be used in class B and viceversa, and, as you said, A and B in our case are independent, they are, let's say, mutually thread safe, so they shouldn't need to share a lock.
Tests may be run in parallel, so we would be adding unnecessary delays in this case. In other similar cases it could have worst implications of course. And yet in other cases you might really need a base class static lock to be shared amongst all inheriting classes.